File: SpringBoot_Part/CreateApi/src/main/java/com/shohab/CreateApi/model/Employee.java
package com.shohab.CreateApi.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Date;

@Data // Generates getters, setters, toString, equals, and hashCode methods
@NoArgsConstructor // Generates a no-argument constructor
@Entity
@AllArgsConstructor
public class Employee  {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private Date joiningDate;
    @Column(unique = true)
    private String email;
    private String gender;
    private String contact;
    private String salary;
    @ManyToOne
    private Department department;

    public Employee(String name, Date joiningDate, String email, String gender, String contact, String salary, Department department) {
        this.name = name;
        this.joiningDate = joiningDate;
        this.email = email;
        this.gender = gender;
        this.contact = contact;
        this.salary = salary;
        this.department = department;
    }



    public Employee(String name) {
        this.name = name;
    }
}

File: PayrollProjectNew/payroll_app/lib/src/features/employee/model/employee.dart
class Employee {
  int? id;
  String? name;
  String? joiningDate;
  String? email;
  String? gender;
  String? contact;
  String? salary;
  Department? department;

  Employee(
      {this.id,
        this.name,
        this.joiningDate,
        this.email,
        this.gender,
        this.contact,
        this.salary,
        this.department});

  Employee.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    name = json['name'];
    joiningDate = json['joiningDate'];
    email = json['email'];
    gender = json['gender'];
    contact = json['contact'];
    salary = json['salary'];
    department = json['department'] != null
        ? new Department.fromJson(json['department'])
        : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['name'] = this.name;
    data['joiningDate'] = this.joiningDate;
    data['email'] = this.email;
    data['gender'] = this.gender;
    data['contact'] = this.contact;
    data['salary'] = this.salary;
    if (this.department != null) {
      data['department'] = this.department!.toJson();
    }
    return data;
  }
}

class Department {
  int? id;
  String? deptName;

  Department({this.id, this.deptName});

  Department.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    deptName = json['deptName'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['deptName'] = this.deptName;
    return data;
  }
}
